src/main/java/core/basesyntax/OperationsStrategy.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public class OperationsStrategy {
+
+    public OperationsStrategy(Fruits fruits) {
+        this.fruits = fruits;
+    }
+
+    private Fruits fruits;
+
+    public int operation(String operation, String fruitType, int amount) {
+        switch (operation) {
+            case "b": // Balance
+                return fruits.balance(fruitType, amount);
+            case "s": // Supply
+                return fruits.supply(fruitType, amount);
+            case "p": // Purchase
+                return fruits.purchase(fruitType, amount);
+            case "r": // Return
+                return fruits.returnFruit(fruitType, amount);
+            default:
+                throw new IllegalArgumentException("Unknown operation: " + operation);

```

Comment:
The error message in the `IllegalArgumentException` could be more descriptive by including the valid operations. For example: `"Unknown operation: " + operation + ". Valid operations are: b, s, p, r."`